Jenkinsfile.revapi

```diff
@@ -0,0 +1,263 @@
+                    assert getMasterRevapiRepos() != ''
+
+                    setRevapiRepos()
+                    checkoutRepo(masterRepos, "master")
+                    checkoutRepo(branched7xRepos, "7.x")
+                }
+            }
+            post {
+                success {
+                    script {
+                        setRepositoriesPropertyIfNeeded('git.branch', getBuildBranch())
+                        setRepositoriesPropertyIfNeeded('git.author', getGitAuthor())
+                        setRepositoriesPropertyIfNeeded('project.version', getRevapiProjectName())
+                    }
+                }
+            }
+        }
+        stage('Prepare for PR'){
+            when {
+                expression { return isCreatePRs() }
+            }
+            steps {
+                prepareForPR()
+            }
+        }
+        stage('Execute Revapi clean-up'){
+            steps {
+                script {
+                    // Execute revapi-clean script and build projects
+                    executeRevapiClean()
+                }
+            }
+        }
+        stage('Build & Test') {
+            when {
+                expression { return !isSkipMavenBuild() }
+            }
+            steps {
+                script {
+                    executeMavenBuild()
+                }
+            }
+        }
+        stage('Create PRs') {
+            when {
+                expression { return isCreatePRs() }
+            }
+            steps {
+                script {
+                    commitAndCreatePR()
+                }
+            }
+        }
+        stage('Merge PR'){
+            when {
+                expression { return isCreatePRs() && isMergePRs() }
+            }
+            steps {
+                mergePR()
+            }
+        }
+    }
+    post {
+        always {
+            script {
+                def propertiesStr = revapiProperties.collect{ entry ->  "${entry.key}=${entry.value}" }.join("\n")
+                writeFile(text: propertiesStr, file: 'revapi-prs.properties')
+                archiveArtifacts(artifacts: 'revapi-prs.properties')
+            }
+            cleanWs()
+        }
+    }
+}
+
+boolean isCreatePRs() {
+    return params.CREATE_PRS
+}
+
+boolean isMergePRs() {
+    return params.MERGE_PRS
+}
+
+boolean isRelease() {
+    return params.RELEASE
+}
+
+boolean isSkipMavenBuild() {
+    return params.SKIP_MAVEN_BUILD
+}
+
+String getGitAuthor(){
+    return params.GIT_AUTHOR
+}
+
+String getBuildBranch(){
+    return params.BUILD_BRANCH_NAME
+}
+
+static String getProjectName(){
+    return "revapi"
+}
+
+String getRevapiBranch(){
+    return "${getRevapiProjectName()}-${env.BOT_BRANCH_HASH}-jenkins"
+}
+
+String getRevapiProjectName() {
+    return "${getProjectName()}-${getRevapiReleaseVersion()}"
+}
+
+String getRevapiReleaseVersion() {
+    return params.REVAPI_RELEASE_VERSION
+}
+
+void setRepositoriesPropertyIfNeeded(String key, def value){
```

Comment:
Maybe call it `setRevapiPropertyIfNeeded`?

Reply:
yes, much better